modules/javafx.base/src/main/java/com/sun/javafx/collections/ObservableSetWrapper.java

```diff
@@ -348,6 +348,16 @@ public boolean removeAll(Collection<?> c) {
     }
 
     private boolean removeRetain(Collection<?> c, boolean remove) {
+        if (c.isEmpty()) {
+            // Retaining an empty set is equivalent to clearing the set.
+            if (!remove && !backingSet.isEmpty()) {
+                clear();
+                return true;
+            }
+
+            return false;
+        }
```

Comment:
This is good, but if `!c.isEmpty()` then we can optimize too, I think: if `backingSet.isEmpty()`, then removing/retaining all will also return `false`.

I think that doing these optimizations in the `removeAll` and `retainAll` before calling this method will be clearer, similarly to how it's done in the `List` case.

Same comment for the `Map` wrapper.

Reply:
+ 1 for this. Otherwise this changes look good to me.